batch7crud-roll1/src/main/java/com/lftechnology/batch7crud/service/EmployeeService.java

```diff
@@ -18,8 +21,14 @@
 
   private EmployeeDAO employeeDAO = new EmployeeDAO();
 
-  public void create(Employee employee) throws DataException {
-    employeeDAO.insert(employee);
+  public void create(Employee employee) throws DataException, ValidationException {   //NOSONAR
+    EmployeeValidator employeeValidator = new EmployeeValidator();
+    Map<String, String> errors = employeeValidator.validate(employee);
+    if (errors.isEmpty()) {
+      employeeDAO.insert(employee);
+    } else {
+      throw new ValidationException(errors);
+    }
```

Comment:
Try to make service class as simple as possible. Throw the exception from validate method.